Packet++/header/GvcpLayer.h

```diff
@@ -0,0 +1,776 @@
+#pragma once
+
+#include "IpAddress.h"
+#include "Layer.h"
+#include "MacAddress.h"
+#include "SystemUtils.h"
+#include <iostream>
+
+/**
+ * @file GvcpLayer.h
+ * @author An-Chi Liu (phy.tiger@gmail.com)
+ */
+
+/**
+ * @namespace pcpp
+ * @brief The main namespace for the PcapPlusPlus lib
+ */
+namespace pcpp
+{
+	namespace internal
+	{
+		static constexpr size_t kGvcpMagicNumber = 0x42;
+		static constexpr size_t kGvcpRequestHeaderLength = 8;
+		static constexpr size_t kGvcpAckHeaderLength = 8;
+		static constexpr size_t kGvcpDiscoveryBodyLength = 248;
+		static constexpr size_t kGvcpForceIpBodyLength = 56;
+	}  // namespace internal
+
+	using GvcpFlag = uint8_t;  // flag bits are specified by each command
+
+	/// @brief Gvcp command
+	/// See spec "18 Command and Acknowledge Values"
```

Comment:
Does this assume the user has the spec?

ditto in the other enums, structs and classes

Reply:
Yes, the user can get a copy of the spec by themselves.

Reply:
Maybe instead of (or in addition to) adding a reference to the spec, we can add some documentation?

Reply:
I agree we don't want to explain the entire protocol in the doxygen documentation, but maybe give just one sentence for each enum / enum value / struct / class?

I don't think we have only a reference to the spec in any other protocol 🤔

Reply:
Sure, for some classes maybe they are worth having a sentence comment. But for most of the enum values, I don't see a good reason to add "a sentence" to explain them as the spec doesn't explain that much as well.

Reply:
We document enum values as well... but if you think it's too much then I'm ok with not documenting enum values. Worst case we can add it later if needed. So let's at least add one sentence for structs/enums/classes?

Reply:
@tigercosmos please address this comment also?